src/workflow.tsx

```diff
@@ -1,12 +1,87 @@
 import * as React from 'react';
-import greet from './greet/greet';
+
+export enum Labels {
+  next = 'next',
+  previous = 'previous',
+}
+
+export interface WorkflowItem {
+  id: string;
+  next?: string;
+  previous?: string;
+  render: React.ReactNode;
+}
 
 interface WorkflowProps {
-  name: string;
+  /** the item that should be displayed when the workflow is rendered */
+  initial?: string;
+  /** the items that the workflow traverses over */
+  items: readonly WorkflowItem[];
 }
 
-const Workflow: React.FunctionComponent<WorkflowProps> = ({ name }) => (
-  <div data-testid="workflow">{greet(name)}</div>
-);
+// construct the path required to get up to, and including, the current item.
+export const buildHistory = (
```

Comment:
What if we extract all helper functions in their own modules to make to workflow component file cleaner, and at the same time, making it easier to test the helper functions in isolation. We would have `buildHistory.ts` and `initializeWorkflow.ts` and their test files along with them (`*.test.ts` files)

Reply:
I am concerned that we would end up with a large number of files/directories that would make navigating the project difficult 🤔 I can see `buildHistory` as a separate module, since it could be used anywhere. `initializeWorkflow` is really just meant to be called like a constructor (not sure how do that in a `FunctionComponent`). I could make that more explicit by making it `private`

Reply:
Yes, from that point of view you're right, we'll end up with more files and directories in that way.  I suppose it's just a matter of preference. What you've said makes perfect sense and I think it shouldn't be that important. In fact, I very often do the same as you did xD. 
Just to share with you, one approach that I've been using and I like when having "utility" functions inside a component file is to put them at the end of the file, so the "main" piece of logic remains at the top. I read about this approach some time ago in [this book](https://ponyfoo.com/books/mastering-modular-javascript) ([specifically in this section](https://github.com/mjavascript/mastering-modular-javascript/blob/master/chapters/ch04.asciidoc#423-an-interdependency-pyramid)). By the way, It is an awesome book. 😁